frontend/packages/console-app/src/components/quick-starts/controller/QuickStartContent.tsx

```diff
@@ -13,52 +14,74 @@ type QuickStartContentProps = {
   onTaskSelect: (selectedTaskNumber: number) => void;
   onTaskReview: (taskStatus: QuickStartTaskStatus) => void;
   onQuickStartChange?: (quickStartId: string) => void;
+  onContentUpdate?: () => void;
 };
 
-const QuickStartContent: React.FC<QuickStartContentProps> = ({
-  quickStart,
-  taskNumber,
-  allTaskStatuses,
-  onTaskSelect,
-  onTaskReview,
-  onQuickStartChange,
-}) => {
-  const {
-    spec: { introduction, tasks, conclusion, nextQuickStart = [] },
-  } = quickStart;
-  const totalTasks = tasks.length;
-  const nextQS = nextQuickStart.length > 0 && nextQuickStart[0];
-  return (
-    <div className="co-quick-start-content">
-      {taskNumber === -1 && (
-        <QuickStartIntroduction
-          tasks={tasks}
-          allTaskStatuses={allTaskStatuses}
-          introduction={introduction}
-          onTaskSelect={onTaskSelect}
-        />
-      )}
-      {taskNumber > -1 && taskNumber < totalTasks && (
-        <QuickStartTasks
-          tasks={tasks}
-          taskNumber={taskNumber}
-          allTaskStatuses={allTaskStatuses}
-          onTaskReview={onTaskReview}
-          onTaskSelect={onTaskSelect}
-        />
-      )}
-      {taskNumber === totalTasks && (
-        <QuickStartConclusion
-          tasks={tasks}
-          conclusion={conclusion}
-          allTaskStatuses={allTaskStatuses}
-          nextQuickStart={nextQS}
-          onQuickStartChange={onQuickStartChange}
-          onTaskSelect={onTaskSelect}
-        />
-      )}
-    </div>
-  );
-};
+type ResizeCallback = () => void;
+
+const QuickStartContent = React.forwardRef(
+  (props: QuickStartContentProps, ref: React.RefObject<HTMLDivElement>) => {
+    const {
+      quickStart,
+      taskNumber,
+      allTaskStatuses,
+      onTaskSelect,
+      onTaskReview,
+      onQuickStartChange,
+      onContentUpdate,
+    } = props;
+    const {
+      spec: { introduction, tasks, conclusion, nextQuickStart = [] },
+    } = quickStart;
+    const totalTasks = tasks.length;
+    const nextQS = nextQuickStart.length > 0 && nextQuickStart[0];
+
+    const resizeCallback: ResizeCallback = () => {
+      ref?.current && onContentUpdate();
+    };
+
+    const debouncedResize = useDebounceCallback(resizeCallback, [resizeCallback], 50);
+
+    React.useEffect(() => {
+      const resizeObserver: ResizeObserver = new ResizeObserver(debouncedResize);
+      resizeObserver.observe(ref.current);
+      return () => {
+        resizeObserver.disconnect();
+      };
+    }, [ref, onContentUpdate, debouncedResize]);
```

Comment:
Why is `onContentUpdate` part of dependency array?

Reply:
it's a lint error otherwise